app/parsers/bulkrax/csv_parser.rb

```diff
@@ -33,16 +33,29 @@ def import_fields
     end
 
     def required_elements?(keys)
-      return if keys.blank?
-      !required_elements.map { |el| keys.map(&:to_s).include?(el) }.include?(false)
+      return unless keys.present?
+      !missing_elements(keys).present?
+    end
+
+    def missing_elements(keys)
+      required_elements - keys.map(&:to_s)
+    end
+
+    def source_identifier
+      @source_identifier ||=  importerexporter.mapping.select do |_, h|
+        h.key?("source_identifier")
+      end.values.first&.[]("from")&.first&.to_sym || :source_identifier
     end
 
     def required_elements
-      %w[title source_identifier]
+      %w[title] + [source_identifier.to_s]
```

Comment:
this will conflict with Alisha's changes. I think there is come confusion about who was doing what. I thought you were going to focus just on the source_idenfier vs system_identifier bit and Alisha would tackle fixing csv parser.

Reply:
using Alisha's version of required elements